Source/WebCore/rendering/RenderFlexibleBox.cpp

```diff
@@ -2409,34 +2409,72 @@ void RenderFlexibleBox::performBaselineAlignment(LineState& lineState)
 {
     ASSERT(lineState.baselineAlignmentState);
 
-    LayoutUnit minMarginAfterBaseline = LayoutUnit::max();
     auto lineCrossAxisExtent = lineState.crossAxisExtent;
     bool containerHasWrapReverse = style().flexWrap() == FlexWrap::Reverse;
-    auto shouldAdjustItemTowardsCrossAxis = [containerHasWrapReverse](ItemPosition alignment) {
-        return (containerHasWrapReverse && alignment == ItemPosition::Baseline) || (alignment == ItemPosition::LastBaseline && !containerHasWrapReverse);
+
+    auto flexItemWritingModeForBaselineAlignment = [&](const RenderBox& flexItem) {
+        if (mainAxisIsChildInlineAxis(flexItem))
+            return flexItem.style().writingMode();
+
+        // css-align-3: 9.1. Determining the Baselines of a Box
+        // In general, the writing mode of the box, shape, or other object being aligned is used to determine
+        // the line-under and line-over edges for synthesis. However, when that writing mode’s block flow direction
+        // is parallel to the axis of the alignment context, an axis-compatible writing mode must be assumed:
+
+        // If the box establishing the alignment context has a block flow direction that is orthogonal to the
+        // axis of the alignment context, use its writing mode.
+        if (style().isRowFlexDirection())
+            return style().writingMode();
+
+        //   Otherwise:
+        //
+        // If the box’s own writing mode is vertical, assume horizontal-tb.
+        // If the box’s own writing mode is horizontal, assume vertical-lr if
+        // direction is ltr and vertical-rl if direction is rtl.
+        if (!flexItem.isHorizontalWritingMode())
+            return WritingMode::HorizontalTb;
+        return style().direction() == TextDirection::LTR ? WritingMode::VerticalLr: WritingMode::VerticalRl;
```

Comment:
```suggestion
        return style().direction() == TextDirection::LTR ? WritingMode::VerticalLr : WritingMode::VerticalRl;
```
Also is it correct to convert sideways writing mode to vertical? if not, it should probably be handled, or this should act based on the BlockFlowDirection.

Reply:
So I *think* this is ok because the block flow direction is equivalent for `vertical-lr/rl` and `sideways-lr/rl` When the spec says:

> If the box’s own writing mode is vertical

I interpreted that as the sideways writing mode being referred to as a "vertical" writing mode  as described here: https://drafts.csswg.org/css-writing-modes-4/#valdef-writing-mode-sideways-rl

I think the way the baseline is synthesized ends up being the same in both cases even with the conversion. Let me know if you think that's not quite correct though....

Reply:
Does the spec text say the baseline is synthesized based on the writing mode or the block flow direction? I'd just follow what the spec says here (I'm also OK doing this in a follow up if it requires a bit of refactoring to use the block flow direction)

Reply:
Yeah this part of the spec that is being implemented here has steps to determine the writing mode specifically